### PR TITLE
Add generator function code style rules

### PR DESCRIPTION
**Pull request by zeppelin**
Since [ember-concurrency](https://github.com/machty/ember-concurrency) has begun using generator functions, it is safe to assume they'll become commonplace across Ember.js projects. This PR is an attempt to declare an officially preferred style rule before the syntax spreads.

The rules currently follow @machty's [choice of style](https://github.com/machty/ember-concurrency/blob/master/tests/unit/task-test.js#L14), but it can be whatever the Core thinks is best.

### CODE REVIEW

**Comment by rwjblue** [comment]
Seems good to me, however I somewhat doubt that we will use generators soonish in Ember itself.

Can you also submit to ember-suave?

File: .jscsrc [hunk 1]
```diff
@@ -87,5 +87,9 @@
     "beforeOpeningRoundBrace": false,
     "beforeOpeningCurlyBrace": true
   },
+  "requireSpacesInGenerator": {
+    "beforeStar": true,
+    "afterStar": true
+  }
```
Comment (rwjblue) [inline .jscsrc]: needs a trailing comma

**Comment by machty** [comment]
Ditto everything @rwjblue said.

**Comment by machty** [comment]
(Also, it's probably too early to say that ember-concurrency will make generator usage commonplace in Ember projects; just want to make it clear that it's still in the very early stages and there are no plans for it to be adopted into ember proper.)

File: .jscsrc (hunk 1 above)
Comment (zeppelin) [inline .jscsrc]: good catch, sorry

**Comment by zeppelin** [comment]
> Can you also submit to ember-suave?

Will do.

> Seems good to me, however I somewhat doubt that we will use generators soonish in Ember itself.

Still better to have it covered early, no? :)

**Comment by rwjblue** [comment]
Closing in favor of the PR to ember-suave (https://github.com/DockYard/ember-suave/issues/102).

**Pull request closed without merging**